templates/components/settings/python/settings.py

```
@@ -113,29 +115,40 @@ def init_azure_openai():
+
 
-    embedding_model = os.getenv("EMBEDDING_MODEL")
-    if embedding_model is None:
-        raise ValueError("EMBEDDING_MODEL environment variable is not set")
+def init_huggingface():
+    try:
+        from llama_index.llms.huggingface import HuggingFaceLLM
+    except ImportError:
+        raise ImportError(
+            "Hugging Face support is not installed. Please install it with `poetry add llama-index-llms-huggingface` and `poetry add llama-index-embeddings-huggingface`"
+        )
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)